docs/CustomUIA.md

```diff
@@ -0,0 +1,125 @@
+<!-- Copyright (c) Microsoft Corporation. All rights reserved.
+     Licensed under the MIT Licence. -->
+
+## Custom UI Automation properties
+Axe Windows supports the inspection of [custom UI Automation properties](https://docs.microsoft.com/en-gb/windows/win32/winauto/uiauto-custompropertieseventscontrolpatterns). The custom properties for which Axe Windows should register are described in a configuration file that is loaded once per session and applies to all inspected UI Automation elements.
+
+### Configuration file format
+Custom UI Automation configuration data is specified in a [JSON](https://en.wikipedia.org/wiki/JSON#Syntax) formatted text file in the configuration directory. The configuration file consists of one object with the following possible attributes:
+
+#### metadata
+This attribute, if specified, can contain values to describe the file as determined by the user. It may be useful, for instance, to store version or authorship information. Example:
+
+```
+  "metadata": {
+    "version": "1.0",
+    "modified": "1970-01-01"
+  }
+```
+
+#### properties
+This attribute contains an array of objects, one per defined custom UI Automation property. Each property object contains the following fields:
+
+##### guid
+The globally unique identifier of this property.
+
+##### programmaticName
+A textual description of this property.
+
+##### uiaType
+The data type of this property's value, one of `"string"`, `"int"`, or `"bool"`.
+
+##### interpretation
+If specified, an object mapping values of a property to more friendly descriptions. This is useful, for instance, when the value of a property is an enumeration.
```

Comment:
Questions:
1. Do we disallow this for non-int types?
2. Should we call out that the property value is the key?
3. Should we call out the behavior if no matching key exists? For example, in the sample, if the input is the number 40, is the output the number 40 or the string "40" ?. It could simplify consumption to have a single type of returned object

Reply:
I've added a bit more context here, but not sure if it's well-explained.